lib/util/git/types.ts

```diff
@@ -9,7 +9,8 @@ export interface GitAuthor {
 
 export type GitNoVerifyOption = 'commit' | 'push';
 
-export type CommitSha = string;
+// We want to make sure it's a long sha of 40 characters and not just a string
+export type LongCommitSha = string & { __longCommitSha: never };
```

Comment:
why not using the old interface name? would case a lot less changes.

Reply:
I opted to make it more explicit, so people would think about it when making a change, if a git call returns a short commit sha, they know they change/convert it to the longer version.  I might add that info to the tsdoc :)

Reply:
Sure, but also causes a lot extra noise inside this PR, should probly done in a seperate PR.

Reply:
Rename done in #25698

Reply:
That was indeed also an option, not doing it seemed strange as it would people just make the same mistake next time again without noticing.
And sorry for not being active anymore, hoping to get to it this weekend to fix your comments and such

Reply:
I made the merge